manager/allocator/networkallocator/networkallocator.go

```diff
@@ -199,6 +193,29 @@ func (na *NetworkAllocator) ServiceAllocate(s *api.Service) (err error) {
 		specNetworks = s.Spec.Networks
 	}
 
+	// Allocate VIPs for all the pre-populated endpoint attachments
+	eVIPs := s.Endpoint.VirtualIPs[:0]
+	var match bool
```

Comment:
The loop below this does something similar, but uses a label and `continue` instead of a flag. I think the two loops should be structured consistently, but I don't have a strong preference about which which approach to use. I find the label/continue approach a little simpler, but I know there are people that don't like it.

If we go with the flag approach, this variable can be defined inside the for loop.